plugin_tooling/src/com/github/rustdt/tooling/RustBuildOutputParserJson.java

```diff
@@ -0,0 +1,254 @@
+ * Copyright (c) 2016 Bruno Medeiros and other Contributors.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Pieter Penninckx - initial implementation
+ *******************************************************************************/
+package com.github.rustdt.tooling;
+
+import java.util.ArrayDeque;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Queue;
+
+import melnorme.utilbox.status.Severity;
+
+import com.eclipsesource.json.Json;
+import com.eclipsesource.json.JsonArray;
+import com.eclipsesource.json.JsonObject;
+import com.eclipsesource.json.JsonValue;
+import com.eclipsesource.json.ParseException;
+
+import melnorme.lang.tooling.toolchain.ops.BuildOutputParser2;
+import melnorme.lang.tooling.toolchain.ops.OperationSoftFailure;
+import melnorme.lang.utils.parse.LexingUtils;
+import melnorme.lang.utils.parse.StringCharSource;
+import melnorme.utilbox.core.CommonException;
+import melnorme.utilbox.misc.IByteSequence;
+import melnorme.utilbox.process.ExternalProcessHelper.ExternalProcessResult;
+
+public abstract class RustBuildOutputParserJson extends BuildOutputParser2 {
+	
+	@Override
+	protected void handleNonZeroExitCode(ExternalProcessResult result) throws CommonException, OperationSoftFailure {
+		// Do nothing
+	}
+	
+	@Override
+	protected IByteSequence getOutputFromProcessResult(ExternalProcessResult result) {
+		return result.getStdErrBytes(); // Use StdErr instead of StdOut.
+	}
+	
+	// One line of Json data can contain more than one ToolMessageData, but we can only return one `ToolMessageData`
+	// from the method `parseMessageData`.
+	// So we create a buffer of remaining data.
+	protected String lineParsing = null; // The line we are currently working with. 
+	protected Queue<ToolMessageData> buffer; // Invariant: if buffer.isEmpty() then lineParsing == null
+	
+	public RustBuildOutputParserJson() {
+		super();
+		this.buffer = new ArrayDeque<ToolMessageData>();
+	}
+	
+	protected static final AbstractRustBuildOutputLineParser CARGO_OUTPUT_LINE_PARSER = 
+			new CargoRustBuildOutputLineParser();
+	
+	@Override
+	protected ToolMessageData parseMessageData(StringCharSource output) throws CommonException {
+		// The call-site stops calling `parseMessageData` when the `output` parameter is exhausted.
+		// So we may not completely exhaust the `output` variable when there are
+		// still `ToolMessageDatas`-s in the buffer.
+		ToolMessageData result = null;
+		try {
+			if (this.lineParsing != null) {
+				// Because of the invariant associated to buffer, we know that this.buffer is not empty.
+				result = this.buffer.remove();
+			} else {
+				this.lineParsing = LexingUtils.stringUntilNewline(output);
+				
+				if (this.lineParsing.startsWith("{")) {
```

Comment:
Ok, I have fully switched to the GSON parsing library, and the associated TODO list is cleared.

It seems like the Travis check has failed because the Travis server could not download some Eclipse stuff (timeout). Is there a way to ask Travis to retry? Or do I have to create an empty dummy commit for that?